celestia/src/common.rs

```diff
@@ -0,0 +1,126 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use celestia_node::network::Network;
+use clap::{error::ErrorKind, ArgGroup, CommandFactory, Parser, ValueEnum};
+use libp2p::multiaddr::Protocol;
+use libp2p::Multiaddr;
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+use crate::{native, server};
+
+const SERVER_DEFAULT_BIND_ADDR: &str = "127.0.0.1:9876";
+
+#[derive(Debug, Parser)]
+// disallow specifying store path when running in browser
+#[clap(group(ArgGroup::new("native_xor_browser")
+             .args(&["store", "browser"])))]
```

Comment:
you can do this instead. I think it's a bit cleaner
```rs
    /// Persistent header store path.
    #[arg(short, long = "store", group = "browser_xor_store")]
    pub(crate) store: Option<PathBuf>,

    /// Serve wasm node which can be accessed with web browser
    #[arg(long, group = "browser_xor_store")]
    browser: bool,
```

Reply:
How does this work? Is `_xor` special?

Reply:
This uses clap grups https://docs.rs/clap/latest/clap/struct.ArgGroup.html. 
By default `multiple` is set to false there, which means that only one argument from the group can be specified and `xor` in the name has not special meaning or use.